### PR TITLE
Add Path/Curve#getTimesAtVectorTangent methods

### PR DESCRIPTION
**Pull request by sasensi**
### Description

I implemented this new methods, as discussed [here](https://github.com/paperjs/paper.js/issues/1513#issuecomment-425647595) to make things like extrusion effect easier.
It will also certainly be usefull for path offset implementation.
As commented in the code, I finally used an algorithm described [here](https://stackoverflow.com/a/34837312/7615922) instead of [this one](https://github.com/paperjs/paper.js/issues/371#issuecomment-277541824) because it already handles edge cases like vertical/horizontal vectors.

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the JSHint rules (`npm run jslint` passes)

### CODE REVIEW

File: src/path/Path.js [hunk 1]
```diff
@@ -2123,6 +2123,42 @@ var Path = PathItem.extend(/** @lends Path# */{
      * the beginning of the path and {@link Path#length} at the end
      * @return {Number} the normal vector at the given offset
      */
+
+    /**
+     * Calculates path-time parameters where the path is tangent to given vector.
```
Comment (lehni) [inline src/path/Path.js]: path-time is a misleading concept that doesn't exist elsewhere in the library. Note that curve-time isn't linear, a curve at t = 0.5 is most of the time no the middle of the curve (at half of its length), because there is tension in curves. path-time here is an emulation where length = 1, 0 = 0, and 0.5 = length / 2.  I think it's better to just speak of lengths instead:

`path.getLengthsWithTangent()`

Comment (sasensi) [reply src/path/Path.js]: Ok, I understand what you say, so I guess it should return a list of lengths instead of a list of "times" between 0 and 1, right ?
And maybe we could speak of offsets instead of lengths as in other path methods ex : [path.getOffsetOf()](http://paperjs.org/reference/path/#getoffsetof-point) ?
`path.getOffsetsWithTangent()`

File: src/path/Curve.js [hunk 2]
```diff
@@ -2289,6 +2247,57 @@ new function() { // Scope for bezier intersection using fat-line clipping
...
+     * is tangential to provided tangent.
+     * Tangents at the start or end are included.
+     *
+     * @param {Number[]} v curve values
+     * @param {Point} point the tangent to which the curve must be tangential
+     * @return {Number[]} at most two curve-time parameters, where the curve is
+     * tangential to the given tangent
+     */
+    function getTimesWithTangent(v, point)
+    {
```
Comment (lehni) [inline src/path/Curve.js]: @sasensi this should be on the same line as `getTimesWithTangent(v, point)` :)

File: src/path/Path.js (hunk 1 above)
Comment (lehni) [reply src/path/Path.js]: Yes you're right! That's the right name for it!

File: src/path/Curve.js (hunk 2 above)
Comment (sasensi) [reply src/path/Curve.js]: Oops, sorry :)

**Comment by lehni** [comment]
@sasensi for future reference: Instead of merging, could you use rebasing so we don't end up with merge commits each time you get the feature branch up to date with the develop branch?

https://www.atlassian.com/git/tutorials/merging-vs-rebasing

**Comment by sasensi** [comment]
@lehni, about rebasing the pull request instead of merging it, Ok, I'll do that next time.
But the reason I did the merge was to trigger travis build to make sure everything is OK before merging the pull request into master. I did it online through github merging interface.
From what I read in this article (https://www.atlassian.com/git/tutorials/merging-vs-rebasing), rebasing once the pull request is done seems to be risky:
> Reviewing a Feature With a Pull Request
If you use pull requests as part of your code review process, you need to avoid using git rebase after creating the pull request. As soon as you make the pull request, other developers will be looking at your commits, which means that it’s a public branch. Re-writing its history will make it impossible for Git and your teammates to track any follow-up commits added to the feature.
Any changes from other developers need to be incorporated with git merge instead of git rebase.
For this reason, it’s usually a good idea to clean up your code with an interactive rebase before submitting your pull request.

How will github pull request interface react if I force push a rebased version of the pull request ?

**Comment by lehni** [comment]
Travis is run every time you push a branch, wether it is rebased or merged doesn't really matter, both result in new commit hashes in the branch.

I changes a few things around after the merge, have a look at f5366fb3cb53bc1ea52e9792e2ec2584c0c4f9c1

Only minor things, mainly to align it with the rest of the code base (and keep lines under 80 chars : )

**Comment by sasensi** [comment]
Ok well I'll try it next time and see by myself.
Thanks for teaching me all those things btw ! 👍

**Comment by sasensi** [comment]
@sapics thanks, I'll consider setting it up on my fork too.
The only drawback I see with that is that you have to push a commit to run tests.
An alternative that allows you to run tests locally on Windows with a closer environment to travis-ci's one is to use `Docker`. I'm actually trying it and I'm pretty close to get everything working.
If you are interested, I can send you, or even as I was saying to @lehni, put somewhere in contributors note, a way to do that.

**Comment by sapics** [comment]
Oh, that's great! Thanks!!
Currently, I have not enough knowledge about Docker,,, so I would like to use it after your `contributors note` :) 

I failed test at very stating poin :_)
`throw new TypeError('"file" argument must be a non-empty string');`
and
`(function (exports, require, module, __filename, __dirname) { ../src/load.js`
`SyntaxError: Unexpected token .`

After I built paper-full.js by gulp, test passes stating point, and I got some errors in TextItem tests which you were reported.
